test/v1/route.go

```diff
@@ -116,9 +117,21 @@ func IsRouteNotReady(r *v1.Route) (bool, error) {
 
 // RetryingRouteInconsistency retries common requests seen when creating a new route
 func RetryingRouteInconsistency(innerCheck spoof.ResponseChecker) spoof.ResponseChecker {
+	const neededSuccesses = 5
```

Comment:
Same as for the thing below basically: We need this downstream because our routers are inconsistent in between themselves too 😂 . So we're adding a retry that checks that we've seen enough succeeding requests in a row to assume all routers are good.

Not sure we want this upstream though, as this problem currently doesn't exist here AFAIK.

On a broader note: We might want to discuss having helpers like the ones we need here upstream and configurable. I could imagine other systems having similar issues if they really want to use the Knative E2E tests on a public service for example, with domains etc all setup.

Wondering if we might want to add flags to control this, i.e. `--ingress-retriable-status-code="404,503" --ingress-consecutive-requests="5"` and Knative upstream's CI just runs with both of these turned off?

Just a thought from the top of my head, but definitely something we might want to discuss and explore I'd think.